src/main/java/de/wejago/hichi2influx/config/MqttClientConfig.java

```diff
@@ -0,0 +1,53 @@
+package de.wejago.hichi2influx.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Slf4j
+@Configuration
+public class MqttClientConfig {
+    private final String PROTOCOL_TCP = "tcp";
+
+    private final String CLIENT_ID="hichi-2-influx-db-application";
+
+    @Value("${mqtt.broker.ip}")
+    private String brokerIp;
+
+    @Value("${mqtt.broker.port}")
+    private String brokerPort;
+
+    @Value("${mqtt.username}")
+    private String username;
+
+    @Value("${mqtt.password}")
+    private String password;
+
+    public IMqttClient setMqttClient() {
```

Comment:
Should be getMqttClient
And with a @Bean annotation, you can even pass it into the Service as a bean, that way it can be mocked easily.

Reply:
I don't get the idea for a **@bean**. Currently the `MqttClientConfig` is a **@Configuration** which makes the whole class a bean. And we can mock it easily.

Reply:
I'll show you later, how I mean it.
You can merge for now.